transforms-caching-experiment/remote-cache-experiment-transforms.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# REQUIRED: Having the key set in the environment is expected - note that remote build cache write is requiered. It can be set here or exported in the terminal before running the script.
+#export DEVELOCITY_ACCESS_KEY=$develocityUrl=<access-key>
+
+gradleVersion=$(./gradlew --version | grep "Gradle" | awk '{print $2}')
+
+major=$(echo "$gradleVersion" | cut -d. -f1)
+minor=$(echo "$gradleVersion" | cut -d. -f2)
+
+# Check if the version is lower than 8.9
+if [ "$major" -lt 8 ] || { [ "$major" -eq 8 ] && [ "$minor" -lt 9 ]; }; then
+  echo "Gradle version $gradleVersion is lower than 8.9"
+  exit 1
+else
+    echo "Gradle version detected: $gradleVersion"
+fi
+
+homeDir=build/HOME
+
+# Set 'task' to the first argument or 'help' if no arguments are provided
+tasks=${*:-help}
+
+# Initialize empty Gradle User Home with settings to run build
+echo "Initializing Gradle User Home directory at $homeDir"
+rm -rf $homeDir
+mkdir -p $homeDir
+mkdir -p $homeDir/caches/"$gradleVersion"/
+cp ~/.gradle/gradle.properties $homeDir
+cp -r ~/.gradle/caches/"$gradleVersion"/generated-gradle-jars $homeDir/caches/"$gradleVersion"/
+
+# Note: This is expecting that CCUD gradle plugin is applied
+export GRADLE_CACHE_REMOTE_PUSH=true
+export GRADLE_CACHE_REMOTE_PATH="cache/$USER-exp-non-task"
```

Comment:
No, not really. The initial run might be faster or slower, but since the only goal of that is to make sure the cache is warmed up it's not really necessary. Using shards is not necessary either, however it's nice to have separation from the "real" cache.